app/controllers/provider_interface/provider_users_invitations_controller.rb

```diff
@@ -106,13 +106,12 @@ def permissions_params
         .permit(provider_permissions: {})
     end
 
-    def require_feature_flag
+    def require_feature_flag!
       render_404 unless FeatureFlag.active?(:providers_can_manage_users_and_permissions)
     end
 
-    def redirect_unless_permitted_to_manage_users
-      can_manage_users = ProviderPermissions.exists?(provider_user: current_provider_user, manage_users: true)
-      render_404 unless can_manage_users
+    def require_manage_user_permission!
+      render_404 unless current_provider_user.authorisation.can_manage_users?
```

Comment:
How about a 403 in this case (user lacks permission)? 404 is still a good fit for feature flags.

Reply:
I think it's good practice to never return 403s, because it reveals the existence of things that the user shouldn't be able to see. In this case it doesn't matter much, but it's best to avoid it altoghether to not accidentally introduce it when it does.

Reply:
We've been using 404s for most things so far and this is fine, but for permissions I wanted to suggest 403s in conjunction with https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/controllers/provider_interface/provider_interface_controller.rb#L40, which explains to the user which permission is missing and the relevant provider, so that users can self-serve (i.e. ask the relevant provider to grant them access).

![image](https://user-images.githubusercontent.com/107591/88167143-45c06080-cc10-11ea-9cb0-4ef9acb062d0.png)

Not a big deal either way.